portal/models/qb_timeline.py

```diff
@@ -359,20 +469,22 @@ def nextQBD(qbds_gen):
         else:
             trace("no trigger date therefore nothing from ordered_qbds()")
             return
+    else:
+        trace("initial trigger date {}".format(td))
 
-    rp_walker = cur_next_rp_gen(user, classification, td)
-    curRPD, nextRPD = next(rp_walker, (None, None))
+    rp_flyweight = RP_flyweight(
+        user=user, trigger_date=td, classification=classification)
 
-    if curRPD:
+    if rp_flyweight.cur_rpd:
```

Comment:
might be cleaner to exit early if no RP found...

Reply:
that would miss the else, where we pull in intervention (i.e. not RP) tied QBs.